extractors/mongodb/extractor_test.go

```diff
@@ -0,0 +1,178 @@
+package mongodb_test
+
+import (
+	"context"
+	"log"
+	"testing"
+	"time"
+
+	"github.com/odpf/meteor/extractors/mongodb"
+	"github.com/stretchr/testify/assert"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+var posts = []interface{}{
+	bson.D{{"title", "World"}, {"body", "Hello World"}},
+	bson.D{{"title", "Mars"}, {"body", "Hello Mars"}},
+	bson.D{{"title", "Pluto"}, {"body", "Hello Pluto"}},
+}
+
+var connections = []interface{}{
+	bson.D{{"name", "Albert"}, {"relation", "mutual"}},
+	bson.D{{"name", "Josh"}, {"relation", "following"}},
+	bson.D{{"name", "Abish"}, {"relation", "follower"}},
+}
+
+var reach = []interface{}{
+	bson.D{{"views", "500"}, {"likes", "200"}, {"comments", "50"}},
+	bson.D{{"views", "400"}, {"likes", "100"}, {"comments", "5"}},
+	bson.D{{"views", "800"}, {"likes", "300"}, {"comments", "80"}},
+}
+
+type Post struct {
+	Title string `bson:"title,omitempty"`
+	Body  string `bson:"body,omitempty"`
+}
+
+type Connection struct {
+	Name     string `bson:"name,omitempty"`
+	Relation string `bson:"relation,omitempty"`
+}
+
+type Reach struct {
+	Views    string `bson:"views,omitempty"`
+	Likes    string `bson:"likes,omitempty"`
+	Comments string `bson:"comments,omitempty"`
+}
+
+func TestExtract(t *testing.T) {
+	t.Run("should return error if no user_id in config", func(t *testing.T) {
+		extractor := new(mongodb.Extractor)
+		_, err := extractor.Extract(map[string]interface{}{})
```

Comment:
let's put password field here to make sure the error is not because of missing password.

Reply:
also host

Reply:
okay!